nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -341,6 +341,33 @@ def run(config):
     address = os.environ.get("REDIS_ADDRESS", config.get("redis_address"))
     ray.init(address=address, local_mode=config.get("local_mode", False))
 
+    # Register serializer and deserializer - needed when logging arrays and tensors.
+    def serializer(obj):
+        if obj.requires_grad:
+            obj = obj.detach()
+        if obj.is_cuda:
+            return obj.cpu().numpy()
+        else:
+            return obj.numpy()
+
+    for tensor_type in [
+        torch.FloatTensor,
+        torch.DoubleTensor,
+        torch.HalfTensor,
+        torch.ByteTensor,
+        torch.CharTensor,
+        torch.ShortTensor,
+        torch.IntTensor,
+        torch.LongTensor,
+        torch.Tensor,
+    ]:
+        def deserializer(serialized_obj):
+            return tensor_type(serialized_obj)  # cast to tensor_type
+
+        ray.register_custom_serializer(
```

Comment:
Due to the extra GPU/CPU transfer involved maybe you make this optional, only register if using `wandb`

Reply:
It seems reasonable to register these by default? This just makes it so that we can log tensors, include them in the result dict, etc. Are you thinking people should get an error when they try to do this, so that they explicitly account for performance?

I'd maybe put it in a separate function so that I can call it if I'm not using `run`.

Reply:
As for registering it only if using wandb: It's not just for that purpose. It's for all logging. I include it with this PR because I came across its benefit as I was testing the wandb integration.

That said, I can add it to a separate function for other use cases.

Reply:
I added the de/serializers to a separate function `register_torch_serializers` in `nupic.research.support.ray_utils.py`.